server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -7471,4 +7474,121 @@ public void testTrimUnsafeCommitHasESVersionInUserData() throws IOException {
             assertThat(userDataAfterTrimUnsafeCommits.get(ES_VERSION), is(equalTo(Version.CURRENT.toString())));
         }
     }
+
+    public void testIndexCommitsListener() throws Exception {
+        final Map<IndexCommit, Engine.IndexCommitRef> acquiredCommits = new HashMap<>();
+        final List<IndexCommit> deletedCommits = new ArrayList<>();
+
+        final Engine.IndexCommitListener indexCommitListener = new Engine.IndexCommitListener() {
+            @Override
+            public void onNewCommit(ShardId shardId, Engine.IndexCommitRef indexCommitRef) {
+                assertThat(acquiredCommits.put(indexCommitRef.getIndexCommit(), indexCommitRef), nullValue());
+                assertThat(shardId, equalTo(shardId));
```

Comment:
```suggestion
                assertThat(shardId, equalTo(InternalEngineTests.this.shardId));
```

Reply:
🤦🏻 Thanks for catching this, I pushed dfd36f5755f27e1f64c61f8ca975bd604728a6cb